File: app/src/main/java/com/example/mediaplayer/chosenSong/ChosenSongViewModelFactory.java
package com.example.mediaplayer.chosenSong;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;
import com.example.mediaplayer.playlist.PlayListModel;
import java.util.ArrayList;

public  class ChosenSongViewModelFactory implements ViewModelProvider.Factory {
    private ArrayList<PlayListModel> playListModels;
    private int chosenSongIndex;

     ChosenSongViewModelFactory( ArrayList<PlayListModel> playListModels, int chosenSongIndex) {
        this.playListModels=playListModels;
        this.chosenSongIndex=chosenSongIndex;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {

        if (modelClass.isAssignableFrom(ChosenSongViewModel.class)) {
            return (T) new ChosenSongViewModel(playListModels,chosenSongIndex);

        }
        throw new IllegalArgumentException("unknown class");
    }
}

File: app/src/main/java/com/example/mediaplayer/chosenSong/ChosenSongViewModelFactory.kt
package com.example.mediaplayer.chosenSong

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.example.mediaplayer.playlist.PlayListModel
import java.util.*

class ChosenSongViewModelFactory(private val playListModels: ArrayList<PlayListModel>?
                                 , private val chosenSongIndex: Int
                                 , private val purposeOfFragment: String?) : ViewModelProvider.Factory {

    override fun <T : ViewModel> create(modelClass: Class<T>): T {

        if (modelClass.isAssignableFrom(ChosenSongViewModel::class.java)) {
            return ChosenSongViewModel(playListModels, chosenSongIndex, purposeOfFragment) as T

        }
        throw IllegalArgumentException("unknown class")
    }
}
